lib/improver/nowcasting/pysteps_advection.py

```diff
@@ -0,0 +1,249 @@
+        if "forecast_reference_time" not in coords:
+            frt_coord = self.analysis_cube.coord('time').copy()
+            frt_coord.rename('forecast_reference_time')
+            self.analysis_cube.add_aux_coord(frt_coord)
+        if "forecast_period" not in coords:
+            self.analysis_cube.add_aux_coord(
+                AuxCoord(np.array([0], dtype=np.int32),
+                         'forecast_period', 'seconds'))
+        # set nowcast attributes
+        self.analysis_cube.attributes['source'] = 'MONOW'
+        self.analysis_cube.attributes['title'] = (
+            'MONOW Extrapolation Nowcast on UK 2 km Standard Grid')
+
+    def _set_up_output_cubes(self, all_forecasts):
+        """
+        Convert 3D numpy array into list of cubes with correct time metadata.
+
+        Args:
+            all_forecasts (np.ndarray):
+                Array of 2D forecast fields returned by extrapolation function
+
+        Returns:
+            forecast_cubes (iris.cube.CubeList):
+                List of extrapolated cubes with correct time coordinates
+        """
+        current_datetime = iris_time_to_datetime(
+            self.analysis_cube.coord('time'))[0]
+        forecast_cubes = [self.analysis_cube.copy()]
+        for i in range(len(all_forecasts)):
+            # copy forecast data into template cube
+            new_cube = self.analysis_cube.copy(
+                data=all_forecasts[i, :, :].astype(np.float32))
+            # update time and forecast period coordinates
+            current_datetime += timedelta(seconds=self.interval*60)
+            current_time = datetime_to_iris_time(current_datetime)
+            new_cube.coord('time').points = np.array(
+                [current_time], dtype=np.int64)
+            new_cube.coord('forecast_period').points = np.array(
+                [(i+1)*self.interval*60], dtype=np.int32)
+            forecast_cubes.append(new_cube)
+        return forecast_cubes
+
+    def _generate_forecast_cubes(self, all_forecasts):
+        """
+        Convert forecast arrays into IMPROVER output cubes with re-added
+        orographic enhancement
+
+        Args:
+            all_forecasts (np.ndarray):
+                Array of 2D forecast fields returned by extrapolation function
+
+        Returns:
+            forecast_cubes (list):
+                List of iris.cube.Cube instances containing forecasts at all
+                required lead times, and conforming to the IMPROVER metadata
+                standard.
+        """
+        # re-mask forecast data
+        all_forecasts = np.ma.masked_invalid(all_forecasts)
+
+        # generate list of forecast cubes
+        self._reformat_analysis_cube()
+        timestamped_cubes = self._set_up_output_cubes(all_forecasts)
+
+        # re-convert units and re-add orographic enhancement
+        forecast_cubes = []
+        for cube in timestamped_cubes:
+            cube.convert_units(self.required_units)
```

Comment:
I can't work out what units ApplyOrographicEnhancement is assuming, I suspect it assumes both precip rate and enhancement data are in the same (SI) units. Perhaps we should convert the orogenh cube to self.required_units at the start of the process function too?

Reply:
The orographic enhancement plugin converts the orographic enhancement into units of the precipitation rate cube: https://github.com/metoppv/improver/blob/master/lib/improver/nowcasting/utilities.py#L175.

Reply:
Good. That's fine then.